src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -53,21 +53,55 @@ protected function render_button(Component\Component $component, RendererInterfa
 			$tpl->touchBlock("disabled");
 		}
 
-		$this->maybe_render_id($component, $tpl);
-
-		return $tpl->get();
+        $this->renderTriggeredComponents($component, $tpl, $default_renderer);
+		$this->maybeRenderId($component, $tpl);
+        return $tpl->get();
 	}
 
-	protected function render_close($component) {
+
+    /**
+     * Renders any components that are triggered by this button
+     * // TODO This functionality does not belong here and should be available in general for any component rendering triggered components
+     *
+     * @param Component\Button\Button $button
+     * @param $tpl
+     * @param $default_renderer
+     * @return Component\Button\Button
+     */
+	protected function renderTriggeredComponents(Component\Button\Button &$button, $tpl, $default_renderer) {
+        if (!count($button->getTriggerActions())) {
+            return $button;
+        }
+        foreach ($button->getTriggerActions() as $action) {
+            $triggered = $action->getComponent();
+            // Hacky: Fake generation of an ID via empty onload code
+            $triggered = $triggered->withOnLoadCode(function($id) {
+                return '';
+            });
+            $triggered_id = $this->bindJavaScript($triggered);
+            $button = $button->withOnLoadCode(function($id) use ($triggered_id, $action) {
+                $event = $action->getEvent();
+                $binding = $action->getJavascriptBinding();
+                $js = $binding($triggered_id);
+                return "$('#{$id}').{$event}(function() { {$js} });";
+            });
+            $tpl->setCurrentBlock('triggered');
+            $tpl->setVariable('COMPONENT', $default_renderer->render($triggered));
```

Comment:
This will only work when the action is triggered at some component that is currently invisible. The component would be rendered twice when it is initially visible.